catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -517,6 +526,112 @@ public void init() {
             return "";
         });
 
+        post("/annotations", (req, res) -> {
+            Map<String, Object> incoming = JsonFactory.create()
+                    .parser()
+                    .parseMap(req.body());
```

Comment:
What happens if the note is larger than your heap size?

Reply:
Yup this is fixed now. After you rebase just make sure to replace any instances of `req.body()` with `util.safeGetBody(req)`